src/devices/Mcp3xxx/Mcp30xx32xx.cs

```diff
@@ -0,0 +1,205 @@
+        /// </summary>
+        /// <param name="positiveChannel">Positive channel to be checked</param>
+        /// <param name="negativeChannel">Negative channel to be checked</param>
+        /// <param name="channelCount">Value representing the number of channels on the device which may vary depending on the configuration.
+        /// If channelCount is not set then the PinCount is used.</param>
+        protected void CheckChannelPairing(int positiveChannel, int negativeChannel, int channelCount = -1)
+        {
+            CheckChannelRange(positiveChannel, (channelCount == -1) ? PinCount : channelCount);
+            CheckChannelRange(negativeChannel, (channelCount == -1) ? PinCount : channelCount);
+
+            // Check that the channel pins are adjacent and part of the same differential pairing
+            if (positiveChannel / 2 != negativeChannel / 2 || positiveChannel == negativeChannel)
+            {
+                throw new ArgumentException($"ADC differential channels must be different and part of the same channel pairing.", nameof(positiveChannel) + " " + nameof(negativeChannel));
+            }
+        }
+
+        /// <summary>
+        /// Reads a  value from the device using pseudo-differential inputs
+        /// </summary>
+        /// <param name="positiveChannel">Channel which represents the signal (valid values: 0 to pincount - 1).</param>
+        /// <param name="negativeChannel">Channel which represents the signal ground (valid values: 0 to pincount - 1).</param>
+        /// <returns>A value corresponding to relative voltage level on specified device channels</returns>
+        public virtual int ReadPseudoDifferential(int positiveChannel, int negativeChannel)
+        {
+            // ensure that the channels are part of the same pairing
+            CheckChannelPairing(positiveChannel, negativeChannel, PinCount / 2);
+
+            return Read(positiveChannel / 2, positiveChannel > negativeChannel ? InputType.InvertedDifferential : InputType.Differential, _adcResolutionBits);
+        }
+
+        /// <summary>
+        /// Reads a value from the device using differential inputs
+        /// </summary>
+        /// <param name="positiveChannel">Channel which represents the positive signal (valid values: 0 to pincount - 1).</param>
+        /// <param name="negativeChannel">Channel which represents the negative signal (valid values: 0 to pincount - 1).</param>
+        /// <returns>A value corresponding to relative voltage level on specified device channels</returns>
+        public virtual int ReadDifferential(int positiveChannel, int negativeChannel)
+        {
+            CheckChannelRange(positiveChannel, PinCount);
+            CheckChannelRange(negativeChannel, PinCount);
+
+            if (positiveChannel == negativeChannel)
+            {
+                throw new ArgumentException($"ADC differential channels must be different.", nameof(positiveChannel) + " " + nameof(negativeChannel));
```

Comment:
it would also be correct to return 0 although it's more likely a bug so makes sense to leave as is :)